provider/terraform/virtual_fields.rb

```diff
@@ -23,6 +23,16 @@ class Terraform < Provider::AbstractCore
     # behaviour. They often don't map to underlying API fields (although they
     # may map to parameters), and will require custom code to be added to
     # control them.
+    #
+    # Virtual fields only support top level, boolean Terraform fields
```

Comment:
Yeah there's no reason you'd ever want a nested virtual field. Being boolean-only (and default-false only) is just because no field has needed it yet.